### PR TITLE
issue/970-streamline-setup

### PR DESCRIPTION
**Pull request by FloLey**
This PR refactors the TS part of the extension and adds/fixes some features

1.fixes some issues with error messages:

- Don’t propose to run “pip install” if no venv is selected, propose to select a venv instead
- Propose to select a venv in this error message shown when no venv is selected.
Peek 2023-03-09 15-51

2. run 'export to server' command in terminal
Peek 2023-03-09 15-42
(Note: the `source /home/florent/.virtualenvs/demovscode/bin/activate` has nothing to do with the extension, It is just run on my system in vscode whenever I open a new terminal)

3. adding the walk through: 
Peek 2023-03-09 17-41
I didn't find an easy way to hide/delete/close the error messages that get resolved while we go through the walk through.

4. links to go to the walk through: those 2 buttons will open the walkthrough.
Screenshot from 2023-03-09 15-20-39
this one is only visible when we are in a .cf file. I created it to debug the command. No idea if it is useful but I didn't find it annoying so I kept it. I can remove it if someone doesn't like it

Screenshot from 2023-03-09 17-45-50
This one gets shown if an error doesn't get resolved.

5. added the installLS/activateLS/projectInstall/openWalkthrough commands.

### CODE REVIEW

File: src/commands.ts [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+
+/**
+ * A function that creates a handler for installing an Inmanta project.
+ * @param {string} pythonPath The path to the Python interpreter.
+ * @returns {() => void} A function that executes the 'inmanta project install' command in a VSCode terminal.
+ */
+export function createProjectInstallHandler(pythonPath: string){
+	return () => {
+		if (!pythonPath || !fileOrDirectoryExists(pythonPath)) {
```
Comment (wouterdb) [inline src/commands.ts:80]: a minor potential improvement is to have one error handling routine, that provides the same recovery path (i.e. always show the install yes/no)

(upon further reading: it already exists)

File: src/language_server.ts [hunk 2]
```diff
@@ -0,0 +1,405 @@
...
+	}
+
+
+	/**
+	 * Propose a solution according to the given error.
+	 *
+	 * @param {LanguageServerDiagnoseResult} error the extension language server error for wich we propose a solution
+	 * @param {string} diagnoseId a uuid to identify the diagnose
+	 */
+	async proposeSolution(error:LanguageServerDiagnoseResult, diagnoseId: string){
```
Comment (wouterdb) [inline src/language_server.ts:99]: nice!

**Review by wouterdb — APPROVED** [review]
1. best documented PR ever
2. it is abundantly clear you actually know typescript, which I don't, so: looks fine to me

File: media/markdown_install_ls.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+### Code navigation
+
+When you hover an element while pressing `Ctrl`, an overview of its Entity type is shown. If you click it (while still having `Ctrl` pressed) you can navigate to the definition of the entity.
+
+![Navigation screenshot](../images/screenshot-ctrl-click.png)
+
+### Compilation error reporting
+
+The extension will run a pre-compilation every time you save your file. Making it easier for you to spot syntax and basic model errors. When hovering the problematic bit, an explaination message is shown, and more details can be found in the **Output** panel (when selecting _Inmanta Server Language_ in the expandable list on the right).
```
Comment (Hugo-Inmanta) [inline media/markdown_install_ls.md:15]:
```suggestion
The extension will run a pre-compilation every time you save your file. Making it easier for you to spot syntax and basic model errors. When hovering the problematic bit, an explanation message is shown, and more details can be found in the **Output** panel (when selecting _Inmanta Server Language_ in the expandable list on the right).
```

File: media/markdown_project_install.md [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+
+Install all modules required for this project.
+
+Clicking on the `Install dependencies` button will start a new terminal and run
+
+```sh
+inmanta project install
+```
+
+More information about the inmanta project command can be found [here](https://docs.inmanta.com/inmanta-service-orchestrator/6/reference/commands.html#install_repeat1).
```
Comment (Hugo-Inmanta) [inline media/markdown_project_install.md:11]:
```suggestion
More information about the `inmanta project` command can be found [here](https://docs.inmanta.com/inmanta-service-orchestrator/latest/reference/commands.html#project).
```

File: package.json [hunk 5]
```diff
@@ -87,16 +107,58 @@
...
             "[inmanta]": {
-              "python.interpreter.infoVisibility": "always"
+                "python.interpreter.infoVisibility": "always"
+            }
+        },
+        "walkthroughs": [
+            {
+                "id": "inmanta.walkthrough",
+                "title": "Get started with the Inmanta extension",
+                "description": "An extenstion to help with the development of Inamnta modules and projects",
```
Comment (Hugo-Inmanta) [inline package.json:117]:
```suggestion
                "description": "This walk-through helps you getting started with the development of Inmanta modules and projects.",
```

File: package.json [hunk 6]
```diff
@@ -87,16 +107,58 @@
...
+                        "title": "Make sure the Language Server is enabled",
+                        "description": "To have the language server start, it needs to be activated in the settings.  \n[Activate Language Server](command:inmanta.activateLS)",
+                        "media": {
+                            "markdown": "media/markdown_activate_ls.md"
+                        }
+                    },
+                    {
+                        "id": "inmanta.projectInstall",
+                        "title": "Install all modules required for the active project",
+                        "description": "Make sure all the missing modules are installed in the development venv. \n[Install dependencies](command:inmanta.projectInstall)",
```
Comment (Hugo-Inmanta) [inline package.json:146]:
```suggestion
                        "description": "Make sure all the missing modules are installed in the development virtual environment. \n[Install dependencies](command:inmanta.projectInstall)",
```

**Review by Hugo-Inmanta — APPROVED** [review]
I did a couple of alternative phrasings suggestions but those are not strong opinions!
I didn't go through the code in detail as I know very little ts, but the overall structure looks nicer  :+1:

**Comment by sanderr** [comment]
As to the commands you added, are they namespaced somehow? Otherwise we may need to add "inmanta" in their names somewhere?

**Review by sanderr — APPROVED** [review]

**Pull request merged**